### PR TITLE
adhere to production ansible-lint profile

### PR DESCRIPTION
**Pull request by artis3n**
This PR fixes ansible-lint errors on the repo.

image

Most of the errors were in test files, so this PR excludes the test directory as syntax requirements aren't necessary in tests, IMO.

That left a few additional issues to fix:

image

Most of _these_ are because `antsibull-changelog`'s yaml changelogs don't adhere to ansible-lint's expectations. These tools can fight it out amongst themselves, and the changelog tool has its own linting check. So the changelog directory is also now excluded from ansible-lint.

This PR addresses the remaining lint errors, leaving ansible-lint happy under the `production` profile, the strongest set of requirements:

image

Contributors can now run pre-commit hooks without failure, and should. Run `make install` or `pre-commit install --install-hooks`.

image

We now also have a PR status check that runs pre-commit to ensure those are upheld in PRs.

### CODE REVIEW

File: playbooks/droplet.yml [hunk 1]
```diff
@@ -9,13 +9,12 @@
...
 
   tasks:
-
     - name: Create SSH key
       digitalocean.cloud.ssh_key:
         state: present
         token: "{{ digitalocean_token }}"
         public_key: "{{ public_key }}"
-        name: "sammy"
+        name: sammy
```
Comment (mamercad) [inline playbooks/droplet.yml:17]: The quotes on this really fail linting?!

**Review by mamercad — APPROVED** [review]
🚢

File: playbooks/droplet.yml (hunk 1 above)
Comment (artis3n) [reply playbooks/droplet.yml:17]: So interestingly, that one didn't fail linting, but running `ansible-lint --fix` auto-did that. 🤷

Comment (artis3n) [reply playbooks/droplet.yml:17]: Why are they making fixes for things they aren't flagging as a linting violation? Is it a violation that didn't populate in the results? 🤷 Ansible tooling is half-madness

File: plugins/modules/firewall.py [hunk 2]
```diff
@@ -83,13 +82,12 @@
         ports: 80
         destinations:
           - addresses:
-            - "0.0.0.0/0"
-            - "::/0"
+              - "0.0.0.0/0"
+              - "::/0"
```
Comment (artis3n) [inline plugins/modules/firewall.py:86]: @mamercad If you notice this spacing is also 4-point spacing when the rest of the yaml is 2-point spacing, why ansible-lint was failing on this I cannot say, but I decided it was easier to acquiesce then question further.

**Pull request merged**